### DISCUSSION TITLE
Using mPDF on WordPress without Access to Composer or Command Line?

### DISCUSSION THREAD

**Discussion by dbareis** [mpdf/mpdf#1959] [General]
Is there a WordPress plugin I can use to install it?  If not has the creation of one been considered? Otherwise I believe WordPress supports its own autoload type feature...

Is there at least a list of what needs to be added?  I have found [Using mPDF without Composer](https://mpdf.github.io/installation-setup/using-without-composer.html) and [composer.json](https://github.com/mpdf/mpdf/blob/master/composer.json) however that is meaningless to me at least :-(    Maybe this is what is the section I need to look at?:

```
"require": {
		"php": "^5.6 || ^7.0 || ~8.0.0 || ~8.1.0 || ~8.2.0 || ~8.3.0",
		"ext-gd": "*",
		"ext-mbstring": "*",
		"mpdf/psr-http-message-shim": "^1.0 || ^2.0",
		"mpdf/psr-log-aware-trait": "^2.0 || ^3.0",
		"myclabs/deep-copy": "^1.7",
		"paragonie/random_compat": "^1.4|^2.0|^9.99.99",
		"psr/http-message": "^1.0 || ^2.0",
		"psr/log": "^1.0 || ^2.0 || ^3.0",
		"setasign/fpdi": "^2.1"
	},
```
However I can't find the mpdf file:
```
> dir "*psr-http-message-shim*" /s /a
 Volume in drive D is D4DATA
 Volume Serial Number is E207-A882
File Not Found
```

I'm new to PHP and downloaded the 8.1.1 zip and took the source directory and tried this code:

```
$DirIncludeMPDF = dirname(__FILE__) . '/PHP Libraries/mPDF/v8.1.0';
set_include_path(get_include_path() . PATH_SEPARATOR . $DirIncludeMPDF);
    require 'Mpdf/Strict.php';
    require 'Mpdf/FpdiTrait.php';
    require 'Mpdf/Mpdf.php';
```

I was hoping to work it out error by error but I'm currently getting "_PHP Fatal error:  Trait "setasign\Fpdi\FpdiTrait" not found in /home/wcipporg/public_html/wp-content/themes/oceanwp-child-theme-master/PHP Libraries/mPDF/v8.1.0/Mpdf/FpdiTrait.php on line 25_" and there is no "setasign\Fpdi" directory.

If I installed Composer on my Windows Computer and installed mPDF would that give me a directory I could FTP to my WordPress site?

**Comment by finwe** [comment]
There are at least a few wordpress plugins using mPDF, it depends on what you want to use mPDF in wordpress for.

> If I installed Composer on my Windows Computer and installed mPDF would that give me a directory I could FTP to my WordPress site?

Yes. You would then just require the created vendor/autoload.php file in your plugin.

**Reply by dbareis** [reply]
Thanks for the response, I was looking for a solution that would allow me to generate PDF files from PHP code.

It's hard to search for plugins, and I have enough of these already and I don't need another thing running in the background.  Now a plugin that just installed mPDF would be another thing.  It would probably make things easier for the other plugin writers as well.

In any case, domPDF was easy to install and does the job well for me so far, if I need to look for something else I'll try Composer on my computer and upload it.
